app/src/org/commcare/gis/EntityMapUtils.kt

```diff
@@ -10,39 +12,163 @@ import org.commcare.utils.AndroidInstanceInitializer
             val address = entity.getFieldString(fieldIndex).trim { it <= ' ' }
             return getLatLngFromAddress(address)
         }
         return null
     }
 
     @Nullable
-    private fun getLatLngFromAddress(address: String): GeoPointData? {
+    private fun getLatLngFromAddress(address: String): LatLng? {
         try {
             if (!address.contentEquals("")) {
                 val data = GeoPointData().cast(UncastData(address))
-                return  data
+                return LatLng(data.latitude, data.longitude)
             }
-        } catch (ignored: IllegalArgumentException) { }
+        } catch (e: IllegalArgumentException) {
+            Logger.exception("Error parsing entity location for map display", e)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityBoundary(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): List<LatLng>? {
+        if (TEMPLATE_FORM_GEO_BOUNDARY == detail.templateForms[fieldIndex]) {
+            val boundaryString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseBoundaryFromString(boundaryString)
+        }
         return null
     }
 
+    @JvmStatic
+    fun getEntityBoundaryColor(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): Int? {
+        if (TEMPLATE_FORM_GEO_BOUNDARY_COLOR == detail.templateForms[fieldIndex]) {
+            val colorString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseHexColor(colorString)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityPoints(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): List<LatLng>? {
+        if (TEMPLATE_FORM_GEO_POINTS == detail.templateForms[fieldIndex]) {
+            val pointListString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parsePointListFromString(pointListString)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityPointColors(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): List<Int>? {
+        if (TEMPLATE_FORM_GEO_POINTS_COLORS == detail.templateForms[fieldIndex]) {
+            val colorsString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseHexColorList(colorsString)
+        }
+        return null
+    }
+
+    @Nullable
+    private fun parseBoundaryFromString(boundaryString: String): List<LatLng>? {
+        try {
+            val parts = boundaryString.trim().split("\\s+".toRegex())
+            val polygon = PolygonUtils.createPolygon(parts)
+            return polygon.map { LatLng(it.latitude, it.longitude) }
+                .toMutableList()
+        } catch (e: IllegalArgumentException) {
+            Logger.exception("Error parsing entity boundary for map display", e)
+            return null
```

Comment:
Think we want to throw any exceptions that might signify incorrect data from here and handle it in map screen by showing a user visible error that indicates something is off about that map they are seeing.

Reply:
>  it most-likely indicates an error on the admin side.

Agree that this would be an app configuration error and the CommCare standard for any app errors is to communicate it out to user as an "Error" in App configuration. 


> How should we show the message? Would a snackbar with an OK button to dismiss be okay? Thinking similar to the new data refresh indicators being added in a different PR right now...

Think this needs to be sticky and non-dismissable error UI. As such a snack bar message that can't be dismissed should be ok here OR just an error text overlapping the map will be sufficient as well. 


> What should the message say?

`There was an error in app configuration due to which not all markers on the map are loaded correctly" .